backend/model/User.js

```diff
@@ -0,0 +1,23 @@
+import mongoose from 'mongoose';
+import bcrypt from 'bcryptjs';
+
+const userSchema = new mongoose.Schema({
+  username: { type: String, required: true, unique: true },
+  password: { type: String, required: true }
+});
+
+userSchema.pre('save', async function(next) {
```

Comment:
In your routes you're using arrow functions, try to be consistent with that throughout the whole code base